drivers/regulator/regulator_common.c

```diff
@@ -28,8 +28,8 @@ int regulator_common_init(const struct device *dev, bool is_enabled)
 	}
 
 	/* regulator voltage needs to be within allowed range before enabling */
-	if ((config->min_uv > INT32_MIN) || (config->max_uv < INT32_MAX)) {
-		ret = regulator_set_voltage(dev, config->min_uv,
+	if ((config->init_uv > INT32_MIN) || (config->max_uv < INT32_MAX)) {
```

Comment:
this now needs rebase (check https://github.com/zephyrproject-rtos/zephyr/pull/54752). Logic should be adjusted with the new code.

I'd treat the new property independently of min/max, just apply set_voltage(dev, init_uv, init_uv) if set in DT. if out of valid range, set_voltage will fail.